packages/cli/test/integration/project.api.test.ts

```diff
@@ -479,261 +479,265 @@ describe('PATCH /projects/:projectId', () => {
+
+			const deleteSpy = jest.spyOn(Container.get(CacheService), 'deleteMany');
+			const resp = await memberAgent.patch(`/projects/${teamProject1.id}`).send({
+				name: teamProject1.name,
+				relations: [
+					{ userId: testUser1.id, role: 'project:admin' },
+					{ userId: testUser3.id, role: 'project:editor' },
+					{ userId: ownerUser.id, role: 'project:viewer' },
+				] as Array<{
+					userId: string;
+					role: ProjectRole;
+				}>,
+			});
+			expect(resp.status).toBe(200);
+
+			expect(deleteSpy).toBeCalledWith([`credential-can-use-secrets:${credential1.id}`]);
+			deleteSpy.mockClear();
+
+			const [tp1Relations, tp2Relations] = await Promise.all([
+				getProjectRelations({ projectId: teamProject1.id }),
+				getProjectRelations({ projectId: teamProject2.id }),
+			]);
+
+			expect(tp1Relations.length).toBe(3);
+			expect(tp2Relations.length).toBe(2);
+
+			expect(tp1Relations.find((p) => p.userId === testUser1.id)).not.toBeUndefined();
+			expect(tp1Relations.find((p) => p.userId === testUser2.id)).toBeUndefined();
+			expect(tp1Relations.find((p) => p.userId === testUser1.id)?.role).toBe('project:admin');
+			expect(tp1Relations.find((p) => p.userId === testUser3.id)?.role).toBe('project:editor');
+			expect(tp1Relations.find((p) => p.userId === ownerUser.id)?.role).toBe('project:viewer');
+
+			// Check we haven't modified the other team project
+			expect(tp2Relations.find((p) => p.userId === testUser2.id)).not.toBeUndefined();
+			expect(tp2Relations.find((p) => p.userId === testUser1.id)).toBeUndefined();
+			expect(tp2Relations.find((p) => p.userId === testUser2.id)?.role).toBe('project:editor');
+			expect(tp2Relations.find((p) => p.userId === ownerUser.id)?.role).toBe('project:editor');
 		});
-		expect(resp.status).toBe(403);
-
-		const [tp1Relations, tp2Relations] = await Promise.all([
-			getProjectRelations({ projectId: teamProject1.id }),
-			getProjectRelations({ projectId: teamProject2.id }),
-		]);
 
-		expect(tp1Relations.length).toBe(2);
-		expect(tp2Relations.length).toBe(2);
-
-		expect(tp1Relations.find((p) => p.userId === testUser1.id)).not.toBeUndefined();
-		expect(tp1Relations.find((p) => p.userId === testUser2.id)).not.toBeUndefined();
-		expect(tp1Relations.find((p) => p.userId === testUser1.id)?.role).toBe('project:viewer');
-		expect(tp1Relations.find((p) => p.userId === testUser2.id)?.role).toBe('project:admin');
-		expect(tp1Relations.find((p) => p.userId === testUser3.id)).toBeUndefined();
-
-		// Check we haven't modified the other team project
-		expect(tp2Relations.find((p) => p.userId === testUser2.id)).not.toBeUndefined();
-		expect(tp2Relations.find((p) => p.userId === testUser1.id)).toBeUndefined();
-		expect(tp2Relations.find((p) => p.userId === testUser2.id)?.role).toBe('project:editor');
-		expect(tp2Relations.find((p) => p.userId === ownerUser.id)?.role).toBe('project:editor');
-	});
-
-	test('should not add from a project adding user with an unlicensed role', async () => {
```

Comment:
Yeah, I don't know what this means, but judging from the body this tests that you can't assign a role that isn't licensed. 

I changed the title and amended the implementation to test editors and viewers.